pkg/apiserver/apiserver.go

```diff
@@ -280,6 +281,15 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
 			return
 		}
 		defer out.Close()
+
+		if wsstream.IsWebSocketRequest(req) {
```

Comment:
We implemented the spdy upgrade check in pkg/apiserver/proxy.go#ServeHTTP(). Is there a reason you put the websocket check here instead of proxy.go?